### PR TITLE
[WebNN EP] Add a duplicate entry to support new "dataType"

### PR DESCRIPTION
**Pull request by Honry**
WebNN spec renames "type" as "dataType" at
https://github.com/webmachinelearning/webnn/pull/464, add a duplicate entry for "dataType" in order to workaround the compatibility issue.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/helper.cc [hunk 1]
```diff
@@ -142,5 +142,43 @@ bool IsValidMultidirectionalBroadcast(std::vector<int64_t>& shape_a,
   return true;
 }
 
+bool SetWebnnDataType(emscripten::val& desc, const int32_t data_type) {
+  // WebNN changed the name of the MLOperandDescriptor's data type from "type" to "dataType",
+  // use a duplicate entry temporarily to workaround this API breaking issue.
+  // TODO: Remove lagecy "type" once all browsers implement the new "dataType".
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/helper.cc:148]: legacy

**Review by fdwr — APPROVED** [review]
Thanks Wanming.

**Pull request merged**